src/google/protobuf/compiler/csharp/csharp_message_field.cc

```diff
@@ -169,7 +154,7 @@ void MessageFieldGenerator::GenerateSerializedSizeCode(io::Printer* printer) {
     printer->Print(
       variables_,
       "if ($has_property_check$) {\n"
-      "  size += $tag_size$ + $tag_size$ + pb::CodedOutputStream.ComputeGroupSize($property_name$);\n"
+      "  size += $tag_size$ + pb::CodedOutputStream.ComputeGroupSize($property_name$);\n"
```

Comment:
Because the changes [over here](https://github.com/protocolbuffers/protobuf/pull/5350#discussion-diff-237694119R60) make tag_size the full tag size instead of half the tag size.